common/environment/environment.cpp

```diff
@@ -966,12 +1004,42 @@ void CLocalEnvironment::buildMachineCache() const
                 x.append(name).append("\"]");
                 Owned<IConstEnvBase> cached = new CConstMachineInfo((CLocalEnvironment *) this, &it->query());
                 cache.setValue(x.str(), cached);
+                if (streq(".", name) || strieq("localhost", name) || streq("127.0.0.1", name) || streq(name, localHostIP.str()))
```

Comment:
The isLocal() itself does not handle 'localhost'. But, the ipset() does. So, I just need to use "if (isLocal())" to replace this line.